### ISSUE TITLE
Numeral Date - Axe violations

### ISSUE THREAD

**Issue by divyajindel** [Sage/carbon#4885]
### Current behaviour

Accessibility Violation - Ensures that every form element has a visible label and is not solely labeled using hidden labels, or the title or aria-describedby attributes.

![image](https://user-images.githubusercontent.com/2328042/156383980-36c59c2f-0047-454c-82c6-80dd3ac77b56.png)

> Ensures all ARIA attributes have valid values
> ARIA attribute element ID does not exist on the page: aria-describedby="d255006b-1a24-6a0c-5e9a-8950d8f1611d-validation-icon"

![image](https://user-images.githubusercontent.com/2328042/156384044-706fb5b4-5a85-47ab-af3e-033f12670bcc.png)

> Ensures that every form element has a visible label and is not solely labeled using hidden labels, or the title or aria-describedby attributes

### Expected behaviour

No Accessibility issue should occur.

### CodeSandbox or Storybook URL

https://carbon.sage.com/?path=/story/numeral-date--size

### Carbon Version

v104.38.2

### What browsers are you seeing the problem on?

Firefox, Chrome, Safari, Microsoft Edge

### What Operating System are you seeing the problem on?

MacOS

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by jamime** [comment]
FE-5012

**Comment by nicktitchmarsh** [comment]
The issues flagged here have been fixed already
